statsmodels/tsa/statespace/sarimax.py

```diff
@@ -0,0 +1,1941 @@
+
+        # Observation covariance matrix
+        if self.measurement_error:
+            self['obs_cov', 0, 0] = params_measurement_variance
+
+        # Transition matrix
+        if self.k_ar > 0 or self.k_seasonal_ar > 0:
+            self[self.transition_ar_params_idx] = reduced_polynomial_ar[1:]
+        elif not self.transition.dtype == params.dtype:
+            self.transition = self.transition.real.astype(params.dtype)
+
+        # Selection matrix (Harvey) or Design matrix (Hamilton)
+        if self.k_ma > 0 or self.k_seasonal_ma > 0:
+            if not self.hamilton_representation:
+                self[self.selection_ma_params_idx] = reduced_polynomial_ma[1:]
+            else:
+                self[self.design_ma_params_idx] = reduced_polynomial_ma[1:]
+
+        # State covariance matrix
+        if self.k_posdef > 0:
+            self['state_cov', 0, 0] = params_variance
+            if self.state_regression and self.time_varying_regression:
+                self[self._exog_variance_idx] = params_exog_variance
+
+        # Initialize
+        if not self._manual_initialization:
+            self.initialize_state()
+
+        return params
+
+
+class SARIMAXResults(MLEResults):
+    """
+    Class to hold results from fitting an SARIMAX model.
+
+    Parameters
+    ----------
+    model : MLEModel instance
+        The fitted model instance
+
+    Attributes
+    ----------
+    params_included : list of str
+        List of parameters actually included in the model, in sorted order.
+    k_ar : int
+        Highest autoregressive order in the model, zero-indexed.
+    k_ar_params : int
+        Number of autoregressive parameters to be estimated.
+    k_diff : int
+        Order of intergration.
+    k_ma : int
+        Highest moving average order in the model, zero-indexed.
+    k_ma_params : int
+        Number of moving average parameters to be estimated.
+    k_seasonal_ar : int
+        Highest seasonal autoregressive order in the model, zero-indexed.
+    k_seasonal_ar_params : int
+        Number of seasonal autoregressive parameters to be estimated.
+    k_seasons : int
+        Number of seasons.
+    k_seasonal_diff : int
+        Order of seasonal intergration.
+    k_seasonal_ma : int
+        Highest seasonal moving average order in the model, zero-indexed.
+    k_seasonal_ma_params : int
+        Number of seasonal moving average parameters to be estimated.
+    k_trend : int
+        Order of the trend polynomial plus one (i.e. the constant polynomial
+        would have `k_trend=1`).
+    k_exog : int
+        Number of exogenous regressors.
+    trend : str{'n','c','t','ct'} or iterable
+        Parameter controlling the deterministic trend polynomial :math:`A(t)`.
+        See the class parameter documentation for more information.
+    model_orders : list of int
+        The orders of each of the polynomials in the model.
+    polynomial_ar : array
+        Array containing autoregressive 
```

Comment:
Actually this reminds me of a question - for the `MLEModel` class, I overrode `from_formula` to throw a `NotImplementedError`, since that doesn't make sense here. Should I do that for these other things like `remove_data`?

At the least, probably I should remove it from the list of methods.